### PR TITLE
Add recipe for black

### PR DESCRIPTION
**Pull request by mrclary**

### CODE REVIEW

File: py2app/recipes/black.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+def check(cmd, mf):
+    m = mf.findNode("black")
+    if m is None or m.filename is None:
+        return None
+
+    egg = Path(get_distribution('black').egg_info)
+    top = egg / 'top_level.txt'
+
+    # These cannot be in zip
+    packages = ["black", "blib2to3"]
```
Comment (ronaldoussoren) [inline py2app/recipes/black.py:14]: Why can't these be in a zipfile? The code doesn't seem to use ``__file__`` (which is a telltale sign for code that won't work in zipfiles), other than to determine if it is running as a compiled C extension.

File: py2app/recipes/black.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    # black may include optimized platform specific C extension which has
+    # unusual name, e.g. 610faff656c4cfcbb4a3__mypyc; best to determine it from
+    # the egg-info/top_level.txt
+    includes = set(top.read_text().strip().split('\n'))
+    includes = list(includes.difference(packages))
+
+    # Missed dependency
+    includes.append('pathspec')
```
Comment (ronaldoussoren) [inline py2app/recipes/black.py:23]: I have to look into this a bit more, the problem here is that py2app performs a regular import when building the dependency graph, and hence picks up the C extension which cannot be scanned for dependencies.

This code is OK for now, but I'll probably research if I can detect if a package uses mypyc and try to scan the corresponding python sources in that case.   That should also take away the need for separate recipes for every package using mypyc.

File: py2app/recipes/black.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from pkg_resources import get_distribution
+from pathlib import Path
```
Comment (ronaldoussoren) [inline py2app/recipes/black.py:2]: The upcoming release of py2app will be the last one that still somewhat supports Python 2.7, and python 2.7 does not support pathlib. Post-release I'll update the distribution metadata for py2app to require at least python 3.6.

That said, if you don't mind that this PR gets pulled after the upcoming release this code is fine (and nicer than code using the older os.path interfaces)

Comment (mrclary) [reply py2app/recipes/black.py:2]: I'd really like this PR to be pulled before the upcoming release, so I'll eliminate `pathlib`.

File: py2app/recipes/black.py (hunk 1 above)
Comment (mrclary) [reply py2app/recipes/black.py:14]: Let me verify and report back the issues.

**Comment by mrclary** [comment]
@ronaldoussoren, I've eliminated `pathlib` and hope that this PR can make it into the next release.

**Comment by ronaldoussoren** [comment]
Thanks, the PR looks good.

**Pull request merged**